MapboxCoreNavigationTests/RouteControllerTests.swift

```diff
@@ -1,116 +1,246 @@
 import XCTest
 import MapboxDirections
 import Turf
+import MapboxMobileEvents
 @testable import MapboxCoreNavigation
 
 fileprivate let mbTestHeading: CLLocationDirection = 50
 
 class RouteControllerTests: XCTestCase {
-    
-    var setup: (routeController: RouteController, firstLocation: CLLocation) {
-        route.accessToken = "foo"
-        let navigation = RouteController(along: route, directions: directions)
-        let firstCoord = navigation.routeProgress.currentLegProgress.nearbyCoordinates.first!
-        return (routeController: navigation, firstLocation: CLLocation(coordinate: firstCoord, altitude: 5, horizontalAccuracy: 10, verticalAccuracy: 5, course: 20, speed: 4, timestamp: Date()))
+
+    struct Constants {
+        static let jsonRoute = (response["routes"] as! [AnyObject]).first as! [String: Any]
+        static let accessToken = "nonsense"
     }
-    
+
+    let eventsManagerSpy = EventsManagerSpy()
+    let directionsClientSpy = DirectionsSpy(accessToken: "garbage", host: nil)
+    let delegate = RouteControllerDelegateSpy()
+
+    lazy var dependencies: (routeController: RouteController, firstLocation: CLLocation, lastLocation: CLLocation) = {
```

Comment:
This is very slick. I might have to start doing this.

Reply:
This pattern existed here previously but wasn't memoized on a var like this. Something to watch out for!